packages/block-editor/src/components/spacing-sizes-control/sides-dropdown/index.js

```diff
@@ -29,7 +29,7 @@ export default function SidesDropdown( {
 		<DropdownMenu
 			icon={ sideIcon }
 			label={ labelProp }
-			className="component-spacing-sizes-control__dropdown"
+			className="components-spacing-sizes-control__dropdown"
```

Comment:
Just realized that we're using `components-*` class names for this component, even if it's not in the `@wordpress/components` package — this could be causing confusion around which package this component belongs to, and in a way "legitimize" using `components-*` classnames outside of the `@wordpress/components` package (which we've been advocating against for some time).

Any changes we could change the `components-*` class name prefix? (of course, it can happen in a separate PR)

Reply:
Good point.

In this case, it was copied from existing controls when the spacing controls were created.

There are a lot of cases where components in packages outside of `@wordpress/components` prefix their component classes with `components-` as well. These include the `editor`, `block-editor`, and `block-library` packages.

The water is muddied further for the spacing controls due to the styling overrides for the range control's marks, etc. 

The tweak commented on here was to make the existing class name at least consistent within the same control. As the spacing sizes control has so far only been exported as experimental from the block-editor package, I think we can change its class names without too much issue.

Other uses of `components-` in class names will need to be evaluated separately as suggested.

Reply:
I've pushed some changes to remove the `components` prefix from the spacing sizes control. Unfortunately, the need to segment the range control does still require the use of some of the range control's internal classes.

Reply:
> The tweak commented on here was to make the existing class name at least consistent within the same control. As the spacing sizes control has so far only been exported as experimental from the block-editor package, I think we can change its class names without too much issue.

Thank you @aaronrobertshaw for the super quick work!

> There are a lot of cases where components in packages outside of `@wordpress/components` prefix their component classes with `components-` as well. These include the `editor`, `block-editor`, and `block-library` packages.

That is absolutely true. It is definitely an uphill battle, especially since only a small group of us are actively trying to bring awareness about it.

It may be hard to remove a lot of those instances outside of the components package, but the least we can do is to avoid introducing new ones and make folks more aware of the high cost that style overrides can have on maintaining our UI components. Your work here is definitely appreciated!

> Other uses of `components-` in class names will need to be evaluated separately as suggested.

Makes sense, let's definitely look at this separately.

Style overrides are in place because we can't achieve the desired result with the available components and their props. When trying to remove those overrides, we should ask ourselves: is the limitation that we're encountering by design? Should we consider updating the component to fulfill this new need? Or should we reconsider what we're working on and use the component how it was intended?

Reply:
>  When trying to remove those overrides, we should ask ourselves: is the limitation that we're encountering by design? 

I'd say for the RangeControl, yes, it is by design, however, the spacing controls being segmented is a pretty left-field design/edge case.

> Should we consider updating the component to fulfill this new need?

We do have the updated SliderControls on the horizon so I don't think need to update a component that should be replaced eventually.

> Or should we reconsider what we're working on and use the component how it was intended?

This wouldn't be possible while meeting the design. So if we don't want the small CSS tweak to the mark positions, we're left with updating the component or creating a "SegmentedRangeControl".

Ultimately, this overriding of the range control's styles isn't introduced in this PR, it's actually lessened by it.

Any follow-ups can make a decision to updating the RangeControl etc.

Reply:
> Ultimately, this overriding of the range control's styles isn't introduced in this PR, it's lessened by it.

Yup! My earlier comment referred to the broader context around this component, and not this PR specifically.

> This wouldn't be possible while meeting the design.

This is very much the point — if we end up in a situation where our designs don't meet the requirements set by other previous design specs, ideally, we should either update the older specs to meet new requirements or update the new design to follow the older specs.

Not a piece of feedback specifically for this PR but more of a general reflection on how I think that, as the developers implementing a design, we should flag this kind of inconsistencies and force them to be resolved one way or the other